src/toxcore/tcp/server/server.rs

```diff
@@ -1237,4 +1267,37 @@ mod tests {
             .wait();
         assert!(handle_res.is_err());
     }
+    #[test]
+    fn tcp_send_pings_test() {
+        let server = Server::new();
+
+        // client #1
+        let (client_1, rx_1) = create_random_client();
+        server.insert(client_1);
+
+        // client #2
+        let (client_2, rx_2) = create_random_client();
+        server.insert(client_2);
+
+        // client #3
+        let (client_3, rx_3) = create_random_client();
+        server.insert(client_3);
+
+        let now = Instant::now();
+
+        let mut enter = tokio_executor::enter().unwrap();
+        // time when all entries is needed to send PingRequest
+        let clock_1 = Clock::new_with_now(ConstNow(
+            now + Duration::from_secs(TCP_PING_FREQUENCY + 1)
+        ));
+
+        with_default(&clock_1, &mut enter, |_| {
+            let sender_res = server.send_pings().wait();
+            assert!(sender_res.is_ok());
+        });
+
+        let (_packet, _rx_1) = rx_1.into_future().wait().unwrap();
```

Comment:
Add check that `ping_id` from packets is the same that is stored inside clients.